lib/glific/tickets.ex

```diff
@@ -73,14 +74,30 @@ defmodule Glific.Tickets do
   """
   @spec create_ticket(map()) :: {:ok, Ticket.t()} | {:error, Ecto.Changeset.t()}
   def create_ticket(attrs \\ %{}) do
-    ticket_params = Map.put_new(attrs, :status, "open")
-
-    with {:ok, ticket} <- do_create_ticket(ticket_params),
+    with {:ok, message_number} <- get_previous_message_number(),
+         {:ok, ticket} <-
+           do_create_ticket(
+             Map.put_new(attrs, :status, "open")
+             |> Map.put_new(:message_number, message_number)
+           ),
          {:ok, _notification} <- create_ticket_notification(attrs) do
       {:ok, ticket}
     end
   end
 
+  @spec get_previous_message_number() :: {:ok, integer()} | {:error, String.t()}
+  defp get_previous_message_number do
+    case Repo.one(from m in Message, order_by: [desc: m.message_number], limit: 1) do
```

Comment:
I don't understand how is this getting the previous message of the contact based on the creation time. This looks like its just fetching the message with the highest message number.